lib/actions-oclif/preload.ts

```diff
@@ -0,0 +1,549 @@
+					exitWithExpectedError(
+						`You have no apps with successful releases for a '${deviceTypeSlug}' device type.`,
+					);
+				}
+				return getCliForm().ask({
+					message: 'Select an application',
+					type: 'list',
+					choices: applications.map((app) => ({
+						name: app.app_name,
+						value: app,
+					})),
+				});
+			},
+		);
+	}
+
+	selectApplicationCommit(releases: Release[]) {
+		const { exitWithExpectedError } = require('../errors');
+
+		if (releases.length === 0) {
+			exitWithExpectedError('This application has no successful releases.');
+		}
+		const DEFAULT_CHOICE = { name: 'current', value: 'current' };
+		const choices = [DEFAULT_CHOICE].concat(
+			releases.map((release) => ({
+				name: `${release.end_timestamp} - ${release.commit}`,
+				value: release.commit,
+			})),
+		);
+		return getCliForm().ask({
+			message: 'Select a release',
+			type: 'list',
+			default: 'current',
+			choices,
+		});
+	}
+
+	async offerToDisableAutomaticUpdates(
+		application: Application,
+		commit: string,
+		pinDevice: boolean,
+	) {
+		const balena = getBalenaSdk();
+
+		if (
+			this.isCurrentCommit(commit) ||
+			!application.should_track_latest_release ||
+			pinDevice
+		) {
+			return;
+		}
+		const message = `\
+
+This application is set to track the latest release, and non-pinned devices
+are automatically updated when a new release is available. This may lead to
+unexpected behavior: The preloaded device will download and install the latest
+release once it is online.
+
+This prompt gives you the opportunity to disable automatic updates for this
+application now. Note that this would result in the application being pinned
+to the current latest release, rather than some other release that may have
+been selected for preloading. The pinned released may be further managed
+through the web dashboard or programatically through the balena API / SDK.
+Documentation about release policies and app/device pinning can be found at:
+https://www.balena.io/docs/learn/deploy/release-strategy/release-policy/
+
+Alternatively, the --pin-device-to-release flag may be used to pin only the
+preloaded device to the selected release.
+
+Would you like to disable automatic updates for this application now?\
+`;
+		return getCliForm()
+			.ask({
+				message,
+				type: 'confirm',
+			})
+			.then(function (update) {
+				if (!update) {
+					return;
+				}
+				return balena.pine.patch({
+					resource: 'application',
+					id: application.id,
+					body: {
+						should_track_latest_release: false,
+					},
+				});
+			});
+	}
+
+	getAppWithReleases(balenaSdk: BalenaSDK, appId: string | number) {
+		return balenaSdk.models.application.get(appId, {
+			$expand: this.applicationExpandOptions,
+		}) as Promise<Application & { should_be_running__release: [Release?] }>;
+	}
+
+	async prepareAndPreload(
+		preloader: Preloader,
+		balenaSdk: BalenaSDK,
+		options: FlagsDef,
+	) {
+		const { ExpectedError } = require('../errors');
```

Comment:
Can you use `import` so we get typings:
```suggestion
		const { ExpectedError } = await import('../errors');
```

Reply:
Ah sorry, missed that.